src/transformStateWithClones.js

```diff
@@ -8,6 +8,30 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const clone = { ...state };
+  const transform = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clone, action.extraData);
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach(key =>
+          delete clone[key]);
+        break;
+      case 'clear':
+        Object.keys(clone).forEach(key =>
+          delete clone[key]);
+        break;
+      default:
+        break;
```

Comment:
I would throw an error like 'Unknown action type'

Reply:
not fixed. Return `default` case